fvm/transactionStorageLimiter.go

```diff
@@ -0,0 +1,82 @@
+package fvm
+
+import (
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type TransactionStorageLimiter struct{}
+
+func NewTransactionStorageLimiter() *TransactionStorageLimiter {
+	return &TransactionStorageLimiter{}
+}
+
+func (d *TransactionStorageLimiter) Process(
+	_ *VirtualMachine,
+	ctx Context,
+	_ *TransactionProcedure,
+	ledger state.Ledger,
+) error {
+	if !ctx.LimitAccountStorage {
+		return nil
+	}
+
+	storageCapacityResolver := ctx.StorageCapacityResolver
+	accounts := state.NewAccounts(ledger)
+
+	registerIds, _ := ledger.RegisterUpdates()
```

Comment:
since we have state now we can simply update this to return a list of account that has changed, so we can iterate over that instead of this logic